libs/server-sdk/src/evaluation/evaluator.cpp

```diff
@@ -65,16 +67,21 @@ EvaluationDetail<Value> Evaluator::Evaluate(
             std::optional<std::size_t> variation_index =
                 detailed_evaluation.VariationIndex();
 
-            // TODO(209589) prerequisite events.
+            event_scope.Send([&](EventFactory const& factory) {
+                return factory.Eval(p.key, context, *descriptor.item,
+                                    detailed_evaluation, Value::Null(),
+                                    flag.key);
+            });
 
             if (!descriptor.item->on || variation_index != p.variation) {
                 return OffValue(flag,
                                 EvaluationReason::PrerequisiteFailed(p.key));
             }
         }
     } else {
-        LogError(parent_key, Error::CyclicPrerequisiteReference(flag.key));
-        return OffValue(flag, EvaluationReason::MalformedFlag());
+        LogError(parent_key.value_or("(no parent)"),
+                 Error::CyclicPrerequisiteReference(flag.key));
+        return EvaluationReason::MalformedFlag();
     }
```

Comment:
This was a bug - it was returning the off variation, but should have instead returned a nullopt.